src/app/core/header/header.component.ts

```diff
@@ -1,22 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { faUser} from '@fortawesome/free-solid-svg-icons';
 import { faSignOutAlt, faSignInAlt } from '@fortawesome/free-solid-svg-icons';
 import { AuthService } from 'src/app/services/auth.service';
 import { Router } from '@angular/router';
 import { COURSES_PATH } from 'src/app/app-routing.module';
+import { Observable, Subscription } from 'rxjs';
+import { User } from '../user-model';
 
 @Component({
   selector: 'vp-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.css']
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit, OnDestroy {
   public faUser = faUser;
   public faSignOutAlt = faSignOutAlt;
   public faSignInAlt = faSignInAlt;
 
+  public userLogin = '';
+
+  subscription: Subscription;
+
   constructor(private authService: AuthService, private router: Router) { }
 
+  ngOnInit() {
+    this.subscription = this.authService.getUserInfo().subscribe(
```

Comment:
you could make userLogin as an observable and use | async in template, it would allow you to remove part of the logic from ts file, no code, nothing to test...